lib/workers/repository/process/lookup/index.ts

```diff
@@ -213,6 +212,15 @@ export async function lookupUpdates(
         return res;
       }
       res.currentVersion = currentVersion!;
+
+      // Reapply package rules to check matches for matchCurrentAge
+      const currentVersionRelease = allVersions.find(
+        (version) => version.version === currentVersion
+      );
+      config = applyPackageRules({
+        ...config,
+        releaseTimestamp: currentVersionRelease?.releaseTimestamp,
+      });
```

Comment:
Why does this need to be done here? Applying package rules is quite "heavy" and this will increase it for every single package

Reply:
We had discussed it https://github.com/renovatebot/renovate/issues/21368#issuecomment-1624764910

From the issue description:
> The "current age" itself should be based on the release timestamps from the datasource. In other words such a package rule can never match until the datasource lookup has completed, but it should then be matched and applied prior to options like separateMajorMinor being evaluated.

separateMajorMinor is evaluated just below here
https://github.com/renovatebot/renovate/blob/c89eeb89dff3ee5a62b71c936bd447e579512e27/lib/workers/repository/process/lookup/index.ts#L255-L257

Reply:
I don't want this feature if it means a performance decrease due to applying packageRules a whole other time on every single lookup. Please explore alternative options

Reply:
I deleted previous comment cause I thought another way was possible, if that is not the cas eI will draft it for now